### PR TITLE
feature/IISCRUM-2548

### PR DESCRIPTION
**Pull request by timyllym**
- use cache to store hostgroup information
- don't do queries if target filters are empty
- don't query for host macros if no hostids are given

### CODE REVIEW

File: src/datasource-zabbix/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts [hunk 1]
```diff
@@ -128,6 +128,18 @@ export class ZabbixAPIConnector {
     return this.request('event.acknowledge', params);
   }
 
+  getGroupsWithHosts() {
+    // Adding random limit to get past Zabbix server cache, this query is cached in datasource instead
```
Comment (sapajune) [inline src/datasource-zabbix/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts:132]: This seems a bit strange. Could you elaborate a bit why this is necessary?

**Review by sapajune — APPROVED** [review]
Looks good overall. 

Just wondering the bit with random limit. IIRC only cache in Zabbix server end that might affect this is the config cache and I don't believe that can be avoided. The configuration cache is just refreshed every 60s by default.

File: src/datasource-zabbix/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts (hunk 1 above)
Comment (timyllym) [reply src/datasource-zabbix/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts:132]: When I about a year ago changed the maintenance status checking to this hostgroup.get API call, I had difficulties in receiving refreshed maintenance_status values from Zabbix server. At that time I added this random limit to the call which seemed to help. But I will check it now again as this query has been changed.

Comment (sapajune) [reply src/datasource-zabbix/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts:132]: Ok. Not entirely sure but I think the maintenance status is part of the config and as such would have this 60s latency which cannot be avoided.

Comment (timyllym) [reply src/datasource-zabbix/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts:132]: For some reason it seems to need the random limit to receive changes in maintenance_status

**Review by bulkware — APPROVED** [review]

**Pull request merged**